wrappers/python/examples/llama_training.py

```diff
@@ -103,114 +93,198 @@
-# num_train_tokens = len(list_train_tokens)
-
-n_positions = 4096
-# num_train_seq = num_train_tokens // (n_positions+1)
-# num_train_batches = num_train_seq // args.batch_size
-# num_train_tokens_truncated = num_train_batches * args.batch_size \
-#         * (n_positions+1)
-# train_tokens = np.array(list_train_tokens[:num_train_tokens_truncated], \
-#         order='F', dtype=np.int64)
-# train_tokens = train_tokens.reshape(num_train_batches, \
-#         num_minibatch, args.minibatch_size, n_positions+1)
-
-input_value = torch.randint(llama_config_torch.vocab_size,
-            (10, n_positions), dtype=torch.int64, device=torch_device)
-position_ids = torch.zeros((10, n_positions)).to(torch.long).to(torch_device)
-
-torch_model.to(torch_device)
-output = torch_model(input_value, position_ids=position_ids)
-print(output.last_hidden_state.shape)
-
-# Transfer input/output to NNTile format
-llama_nntile.activations[0].value.from_array(input_value.T.cpu().numpy())
-llama_nntile.forward_async()
-nntile.starpu.wait_for_all()
-llama_nntile_output_np = np.zeros((4096, 4096, 10), order="F",
-                                  dtype=np.float32)
-llama_nntile.activations[-1].value.to_array(llama_nntile_output_np)
-
-output_torch = output.last_hidden_state.cpu().detach().numpy()
-print(output_torch.shape, llama_nntile_output_np.shape,
-      llama_nntile_output_np.T.shape)
-print(np.linalg.norm(output_torch - llama_nntile_output_np.T) /
-      np.linalg.norm(output_torch))
-
-llama_nntile.unregister()
+assert args.nepochs >= 0
+
+# Load named pretrained PyTorch model
+if args.pretrained == "remote":
+    # Newer versions of transformers can use fast attention, so we disable it
+    # through a parameter attn_implementation
+    model_torch = LlamaForCausalLM.from_pretrained(args.remote_model_name,
+                cache_dir=args.model_path, local_files_only=True)
+elif args.pretrained == "local":
+    if args.config_path:
+        f = open(args.config_path)
+        conf_dict = json.load(f)
+        f.close()
+        config = LlamaConfig(**conf_dict)
+        model_torch = LlamaForCausalLM(config)
+        tokenizer = None
+        if args.optimizer == "adam":
+            optimizer = Adam(model_torch.parameters(), args.lr)
+        elif args.optimizer == "sgd":
+            optimizer = SGD(model_torch.parameters(), args.lr)
+        elif args.optimizer == "adamw":
+            optimizer = AdamW(model_torch.parameters(), args.lr)
+        else:
+            raise ValueError
+        if args.checkpoint_path:
+            checkpoint = torch.load(args.checkpoint_path)
+            model_torch.load_state_dict(checkpoint['model_state_dict'])
+            optimizer.load_state_dict(checkpoint['optimizer_state_dict'])
+
+tokenizer = LlamaTokenizerFast.from_pretrained(args.remote_model_name,
+                                                cache_dir=args.model_path)
+model_torch.eval()
+print(model_torch.config)
+
+if args.nepochs > 0:
```

Comment:
it makes no sense to pass `nepochs=0` as all the other parameters like `nforward` are removed from this script. I think we need to set up `nepochs=1` as a default value.